src/pkgcheck/checks/metadata.py

```diff
@@ -733,11 +733,15 @@ class DependencyCheck(Check):
 
     def __init__(self, *args, use_addon):
         super().__init__(*args)
-        self.deprecated = self.options.target_repo.deprecated.match
         self.iuse_filter = use_addon.get_filter()
         self.conditional_ops = {'?', '='}
         self.use_defaults = {'(+)', '(-)'}
 
+        restrictions = set()
+        for repo in self.options.target_repo.trees:
+            restrictions.update(repo.deprecated.restrictions)
+        self.deprecated = boolean.OrRestriction(*restrictions, finalize=True).match
+
```

Comment:
I'd move this inherited support directly into pkgcore under ebuild repo objects. See https://github.com/pkgcore/pkgcore/commit/8b3b21b2426aa87ca7f9bd32bd66657dcb047cb3 as an example of what I'm talking about.